tt_metal/impl/dispatch/device_command.hpp

```diff
@@ -368,7 +368,6 @@ class DeviceCommand {
         }
     }
     void add_dispatch_set_unicast_only_cores(const std::vector<uint32_t>& noc_encodings, DispatcherSelect dispatcher_type) {
-        TT_ASSERT(noc_encodings.size());
```

Comment:
so the fix is to just go in here and do nothing for N150 and GS? Maybe add a comment? Or maybe assert that for GS and N150 that this is empty? It just seems a little sus that this is silently doing nothing in the background.

Reply:
note that this code will get much more clear w/ the hal in place (I think)

Reply:
Added comment explaining the implicit behaviour here